js-packages/@prestojs/ui-antd/src/getWidgetForField.ts

```diff
@@ -99,10 +101,19 @@ export default function getWidgetForField<
     | null {
     const { fieldClassName } = Object.getPrototypeOf(field).constructor;
     // Couldn't work out what to type this as so field.constructor was accepted
-    const widget: FieldWidgetType<any, any> | null | undefined =
-        field.choices || field.asyncChoices
-            ? choicesMapping.get(fieldClassName) || mapping.get(fieldClassName)
-            : mapping.get(fieldClassName);
+    const getWidget = (clsName: string): FieldWidgetType<any, any> | null | undefined => {
+        if (clsName === 'ListField') {
+            if (field.choices) {
+                return React.lazy(() => import('./widgets/SelectChoiceWidget'));
+            } else {
```

Comment:
this doesn't seem safe - why the assumption that this is `asyncChoices`? This looks like it will break if it's not, eg. `new ListField({ childField: new IntegerField() })`

Reply:
what should be the default for ListField you think? we were not catering for it before - shall we just raise if both choices & asyncChoices are missing?